fabric-convention-tags-v2/src/main/java/net/fabricmc/fabric/api/tag/convention/v2/ConventionalBiomeTags.java

```diff
@@ -0,0 +1,155 @@
+	public static final TagKey<Biome> IS_WINDSWEPT = register("is_windswept");
+	public static final TagKey<Biome> IS_JUNGLE = register("is_jungle");
+	/**
+	 * For temperate or warmer plains-like biomes.
+	 * For snowy plains-like biomes, see {@link ConventionalBiomeTags#IS_SNOWY_PLAINS}.
+	 */
+	public static final TagKey<Biome> IS_PLAINS = register("is_plains");
+	public static final TagKey<Biome> IS_SAVANNA = register("is_savanna");
+	/**
+	 * For land biomes where ice naturally spawns.
+	 * For biomes where snow alone spawns, see {@link ConventionalBiomeTags#IS_SNOWY}.
+	 */
+	public static final TagKey<Biome> IS_ICY = register("is_icy");
+	/**
+	 * For water biomes where ice naturally spawns.
+	 * For biomes where snow alone spawns, see {@link ConventionalBiomeTags#IS_SNOWY}.
+	 */
+	public static final TagKey<Biome> IS_AQUATIC_ICY = register("is_aquatic_icy");
+	/**
+	 * Biomes that exist on the shoreline of a body of water.
+	 */
+	public static final TagKey<Biome> IS_BEACH = register("is_beach");
+	/**
+	 * Biomes densely populated with deciduous trees.
+	 */
+	public static final TagKey<Biome> IS_FOREST = register("is_forest");
+	public static final TagKey<Biome> IS_BIRCH_FOREST = register("is_birch_forest");
+	public static final TagKey<Biome> IS_OCEAN = register("is_ocean");
+	public static final TagKey<Biome> IS_DESERT = register("is_desert");
+	public static final TagKey<Biome> IS_RIVER = register("is_river");
+	public static final TagKey<Biome> IS_SWAMP = register("is_swamp");
+	public static final TagKey<Biome> IS_MUSHROOM = register("is_mushroom");
+	public static final TagKey<Biome> IS_UNDERGROUND = register("is_underground");
+	public static final TagKey<Biome> IS_MOUNTAIN = register("is_mountain");
+
+	public static final TagKey<Biome> IS_HOT = register("is_hot");
+	public static final TagKey<Biome> IS_HOT_OVERWORLD = register("is_hot/overworld");
+
+	public static final TagKey<Biome> IS_TEMPERATE = register("is_temperate");
+	public static final TagKey<Biome> IS_TEMPERATE_OVERWORLD = register("is_temperate/overworld");
+
+	public static final TagKey<Biome> IS_COLD = register("is_cold");
+	public static final TagKey<Biome> IS_COLD_OVERWORLD = register("is_cold/overworld");
+
+	public static final TagKey<Biome> IS_WET = register("is_wet");
+	public static final TagKey<Biome> IS_WET_OVERWORLD = register("is_wet/overworld");
+
+	public static final TagKey<Biome> IS_DRY = register("is_dry");
+	public static final TagKey<Biome> IS_DRY_OVERWORLD = register("is_dry/overworld");
+
+	public static final TagKey<Biome> IS_VEGETATION_SPARSE = register("is_sparse_vegetation");
+	public static final TagKey<Biome> IS_VEGETATION_SPARSE_OVERWORLD = register("is_sparse_vegetation/overworld");
+
+	public static final TagKey<Biome> IS_VEGETATION_DENSE = register("is_dense_vegetation");
+	public static final TagKey<Biome> IS_VEGETATION_DENSE_OVERWORLD = register("is_dense_vegetation/overworld");
```

Comment:
Are these specific tags really necessary? I usually use this and it works
```java
if (biome.is(ConventionalBiomeTags.HOT) && biome.is(ConventionalBiomeTags.OVERWORLD)) {
    // ...
}
```
I understand that this is a very convenient shorthand, but why do only these tags have it? I believe that merging two tags should not happen at all.

Sorry if you already answered this question and I missed it

Reply:
It comes from how forge (and now neo) have folder biome tags that holds dimensions for those that want to target a specific dimension of a category. For example, cold nether biomes in their structure json file which can be tricky to do tag intersections with depending on loader. 

It was also ported to the fabric pr so the two are synced up so that I don’t end up deleting biome tags from the Neo side that could be in use
https://github.com/neoforged/NeoForge/tree/1.20.x/src/generated/resources/data/forge/tags/worldgen/biome

is_dense was the original name on Neo side but that was too vague so I combined that name with vegetation name fabric had to get is_vegetation_dense folder that’s clear and then dimensions inside so that people can tag their biomes for easier usage by other.

Might also help modded dimensions? Like twilight forest to have a tag under is_cold for their cold biomes so that datapacks using is_cold/overworld tag in their structure json doesn’t spawn the structure in twilight forest by mistake. 

I guess it would also boil down to, does fabric have a system that changes the biome holder set in the structure json file to allow tag intersections?

Reply:
Just asked around, Fabric does not have custom RegistryEntryLists so the Biome field in structure JSONs cannot do tag intersections.
![image](https://github.com/FabricMC/fabric/assets/40846040/273a4d71-a05e-41a9-b110-9cd8a0ad926f)

If Fabric adds custom RegistryEntryLists like they did with custom ingredients for recipes, then I'll reconsider these folders for individual dimensions. But for now, we should keep these category/dimension combo tags for best usability for structure JSONs which is the most common consumer of these biome tags.

I'll mark this as resolved and can revisit if custom RegistryEntryLists is added by chance in future.

Reply:
> If Fabric adds custom RegistryEntryLists
I suggest you open the issue and leave a link to these comments 👍

Reply:
Here you go: https://github.com/FabricMC/fabric/issues/3372